src/lib/asn1/asn1_oid.cpp

```diff
@@ -152,13 +152,22 @@ bool OID::registered_oid() const {
    return !human_name_or_empty().empty();
 }
 
-size_t OID::hash_code() const {
-   constexpr uint64_t mod = 0xffffffffffffffc5;
-   uint64_t hash = 0;
+bool OID::matches(std::initializer_list<uint32_t> other) const {
+   if(m_id.size() == other.size()) {
+      return std::equal(m_id.begin(), m_id.end(), other.begin());
+   } else {
+      return false;
+   }
```

Comment:
Since C++17 there's another overload for `std::equal` that includes the length check:

```suggestion
   // TODO: once all target compilers support it, use std::ranges::equal
   return std::equal(m_id.begin(), m_id.end(), other.begin(), other.end());
```

Reply:
TIL thanks